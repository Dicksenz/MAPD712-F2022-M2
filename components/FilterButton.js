import React from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";

const FilterButton = ({ title, onPress, selectedIndex, index }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={{
          ...styles.container,
          backgroundColor: selectedIndex === index ? "#86B1B6" : "white",
        }}
      >
        <Text
          style={{
            ...styles.textStyle,
            color: selectedIndex === index ? "white" : "#A9A4A4",
          }}
        >
          {title}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 25,
    paddingRight: 25,
    borderRadius: 20,
    borderColor: "#86B1B6",
    borderWidth: 1,
  },

  textStyle: {
    color: "white",
    fontSize: 17,
  },
});

export default FilterButton;
