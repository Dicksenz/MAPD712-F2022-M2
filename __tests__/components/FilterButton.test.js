import React from "react";
import renderer from "react-test-renderer";
import FilterButton from "../../components/FilterButton";

test("Check if CustomButton component renders correctly", () => {
  const tree = renderer.create(<FilterButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
