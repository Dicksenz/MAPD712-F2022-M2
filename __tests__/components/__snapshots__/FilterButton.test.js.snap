// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if CustomButton component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#86B1B6",
        "borderColor": "#86B1B6",
        "borderRadius": 20,
        "borderWidth": 1,
        "paddingBottom": 10,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 17,
        }
      }
    />
  </View>
</View>
`;
